### DISCUSSION TITLE
How many versions of backwards compatibility is wise?

### DISCUSSION THREAD

**Discussion by DavyLandman** [microsoft/vscode-discussions#1484] [Extension Development QnA] [answered]
Are there any statistics about the pace that users pickup new versions of VS Code? For example, [1.90 moved to node 20](https://code.visualstudio.com/updates/v1_90#_electron-29-update), quite some of the dependencies in our extension have also jumped to node 20. But as soon as we switch to node 20, we should mark our extension to require at least vs code 1.90.  While right now we require 1.82 (since that's the one that jumped to node 18).

So my question is, what is wisdom here, how long do people keep backwards compatibility in their extensions?

I've also though about allowing node 20 for the `devDependencies` but keeping a close watch on the normal dependencies (of which we only have a few), but that does become a bit more brittle, aka, harder to notice if we accidentally break node 18 compatibility by an upgrade to a library that doesn't run in node 18.

**Comment by isidorn** [comment, marked as answer]
Hi VS Code PM here 👋 
I suggest 3-4 versions (not including recovery releases). This is just my take. We try to auto-update users aggressively so they are always on latest version.
